runtime/src/lib.rs

```diff
@@ -128,37 +133,48 @@ impl Executor {
 		Executor { inner: Mode::ThreadPerFuture }
 	}
 
-	/// Spawn a future on this runtime
+	/// Spawn a legacy future on this runtime
 	pub fn spawn<R>(&self, r: R)
 	where
-		R: IntoFuture<Item = (), Error = ()> + Send + 'static,
-		R::Future: Send + 'static,
+		R: Future01<Item = (), Error = ()> + Send + 'static,
```

Comment:
what are the benefits of this (breaking) change or the argument from `impl IntoFuture` to `impl Future`?